lte/gateway/c/session_manager/SessionManagerServer.cpp

```diff
@@ -61,23 +61,36 @@ void LocalSessionManagerAsyncService::init_call_data() {
 
 SessionProxyResponderAsyncService::SessionProxyResponderAsyncService(
     std::unique_ptr<ServerCompletionQueue> cq,
-    std::unique_ptr<SessionProxyResponderHandler> handler)
-    : AsyncService(std::move(cq)), handler_(std::move(handler)) {}
+    std::shared_ptr<SessionProxyResponderHandler> handler)
+    : AsyncService(std::move(cq)), handler_(handler) {}
 
 void SessionProxyResponderAsyncService::init_call_data() {
   new ChargingReAuthCallData(cq_.get(), *this, *handler_);
   new PolicyReAuthCallData(cq_.get(), *this, *handler_);
```

Comment:
Each 'AsyncService' corresponds to a GRPC service. So I can't easily do this without changing the signature of the asyn grpc handling we have. (which is not trivial haha)
And the AbortSession service is separate from the SessionProxy GRPC service, since it is used in AAA on its own.